src/interrogate/badge_gen.py

```diff
@@ -110,16 +113,14 @@ def save_badge(
 
     # need to write the badge as an svg first in order to convert it to
     # another format
-    tmp_output_file = f"{os.path.splitext(output)[0]}.tmp.svg"
+    tmp_output_file = Path(f"{output.parent / output.stem}.tmp.svg")
```

Comment:
```suggestion
    tmp_output_file = output.with_suffix(".tmp.svg")
```
Source: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix